docs/dev/building-on-zksync/contracts/differences-with-ethereum.md

```diff
@@ -31,22 +32,22 @@ It is advised not to use the former, but rather to transform them into the follo
 (bool s, )= call{value: x}("")
 ```
 
+This converts the .send/.transfer functionality to call while avoiding potential security risks.
+
 ### Non-standard `CREATE`/`CREATE2` behaviour
 
-In Ethereum, the `initCode` of a contract with concatenated constructor parameters serves as the input for both `CREATE` and `CREATE2`.
- In contrast, on the zkSync Era, contract deployment occurs through the use of hashes. The operator receives the bytecodes in the `factoryDeps` fields of the EIP712 transactions while the actual [deployment](./contract-deployment.md#format-of-bytecode-hash) process involves providing the contract's hash to the `ContractDeployer` system contract.
+On Ethereum, both `CREATE` and `CREATE2` accept the `initCode` of the contract with constructor parameters concatenated to it as its input. However, on zkSync, contract deployment is done via hashes, and the factoryDeps fields of the EIP712 transactions contain the bytecodes themselves. The actual deployment is done by providing the contract’s hash to the `ContractDeployer` system contract.
 
+To ensure that `create`/`create2` works correctly, the compiler must know the bytecode of the deployed contract beforehand. The compiler interprets the arguments of create as slightly unfilled arguments calldata to the ContractDeployer, and the datasize/ dataoffset Yul instructions were adapted to return a constant size and the bytecode hash instead of bytecode.
```

Comment:
```suggestion
To ensure that `create`/`create2` works correctly, the compiler must know the bytecode of the deployed contract beforehand. The compiler interprets the arguments of `create` as slightly unfilled arguments calldata to the ContractDeployer, and the datasize/ dataoffset Yul instructions were adapted to return a constant size and the bytecode hash instead of bytecode.
```

Reply:
**unfilled arguments calldata** ??

Reply:
**a constant size** ??

Reply:
Changed in a push